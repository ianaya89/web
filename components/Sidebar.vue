<template>
  <div class="sidebar-nav-column hidden xl:flex bg-white">
    <nav class="fixed" style="max-height: 100vh">
      <slot />
    </nav>

    <div
      class="sidebar__footer text-grey-l1 px-8 py-4 flex flex-col items-start justify-center fixed bottom-0"
      style="width: 12.5rem;"
    >
      <slot name="footer" />
    </div>
  </div>
</template>

<script>
export default {}
</script>

<style lang="postcss" scoped>
.sidebar-nav-column {
  @apply .border-r .border-grey-l3 .py-4 .flex-wrap .flex-shrink-0;
  width: 12.5rem;
  min-height: calc(100vh - 5rem);

  .link--secondary-small {
    @apply .font-sans .text-p3 .text-grey-l1;

    &:hover,
    &:active {
      @apply .text-grey-l2 .no-underline .border-b-0;
    }
  }

  .sidebar__footer {
    .nuxt-link-exact-active {
      @apply .text-red;
    }
  }
}
</style>
