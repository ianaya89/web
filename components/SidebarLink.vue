<template>
  <nuxt-link
    :to="localePath(href)"
    class="menu-item text-h7 xl:text-menu mb-6 xl:mb-3"
    style="width: 12.5rem;"
    active-class="active"
  >
    <div class="item-content">
      <slot />
      <badge v-if="badge" :value="badgeValue" />
    </div>
    <div class="item-lock">
      <i v-if="locked" class="fa fa-fw fa-lock" />
    </div>
  </nuxt-link>
</template>

<script>
import Badge from '@/components/Badge'

export default {
  components: {
    Badge
  },

  props: {
    href: {
      type: Object,
      required: true
    },
    badge: {
      type: Boolean,
      default: false
    },
    badgeValue: {
      type: Number,
      default: null
    },
    locked: {
      type: Boolean,
      default: false
    }
  }
}
</script>

<style lang="postcss">
.menu-item {
  @apply .text-black .font-sans .font-semibold .uppercase .tracking-wide .flex .mt-1 .px-6 .pt-2 .pb-1 .border-l-3 .border-transparent;
  transition: 150ms ease-in-out;
  .item-content {
    flex-grow: 1;
  }

  &:hover {
    @apply .text-grey .no-underline .border-b-0;
  }

  &.active {
    @apply .text-red .border-l-3 .border-b-0 .border-red;
  }
}
</style>
