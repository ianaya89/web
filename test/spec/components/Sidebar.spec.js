import { mount, shallowMount } from '@vue/test-utils'
import Sidebar from '@/components/Sidebar'
import SidebarLink from '@/components/SidebarLink'

describe('Sidebar', () => {
  it('renders default empty sidebar', () => {
    const wrapper = shallowMount(Sidebar)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders sidebar with links', () => {
    const linkWrapper = {
      render(render) {
        return render(SidebarLink, {
          propsData: {
            href: {
              name: 'team-profile-fans'
            }
          },
          slots: {
            default: 'Profile fans link'
          }
        })
      }
    }
    const linkWrapper2 = {
      render(render) {
        return render(SidebarLink, {
          propsData: {
            href: {
              name: 'team-dashboard'
            }
          },
          slots: {
            default: 'Dashboard'
          }
        })
      }
    }

    const wrapper = mount(Sidebar, {
      slots: {
        default: [linkWrapper, linkWrapper2]
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
