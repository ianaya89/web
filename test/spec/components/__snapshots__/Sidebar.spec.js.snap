// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders default empty sidebar 1`] = `
<div
  class="sidebar-nav-column hidden xl:flex bg-white"
>
  <nav
    class="fixed"
    style="max-height: 100vh;"
  />
   
  <div
    class="sidebar__footer text-grey-l1 px-8 py-4 flex flex-col items-start justify-center fixed bottom-0"
    style="width: 12.5rem;"
  />
</div>
`;

exports[`Sidebar renders sidebar with links 1`] = `
<div
  class="sidebar-nav-column hidden xl:flex bg-white"
>
  <nav
    class="fixed"
    style="max-height: 100vh;"
  >
    <a
      class="menu-item text-h7 xl:text-menu mb-6 xl:mb-3"
      style="width: 12.5rem;"
    />
    <a
      class="menu-item text-h7 xl:text-menu mb-6 xl:mb-3"
      style="width: 12.5rem;"
    />
  </nav>
   
  <div
    class="sidebar__footer text-grey-l1 px-8 py-4 flex flex-col items-start justify-center fixed bottom-0"
    style="width: 12.5rem;"
  />
</div>
`;
